src/arrayMethodJoin.js

```diff
@@ -4,8 +4,29 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let string = '';
+
+    for (let i = 0; i < this.length; i++) {
+      if (this[i] === null || this[i] === undefined) {
+        string += separator;
+        continue;
+      }
+
+      string += this[i];
+
+      if (i < this.length - 1) {
+        string += separator;
+      }
+    }
+
+    const lastElement = this[this.length - 1];
+
+    if (lastElement === undefined || isNaN(lastElement)) {
+      string = string.slice(0, -1);
+    }
```

Comment:
why do you need it? if the last element is, for example `'a'` you will remove it?
![image](https://user-images.githubusercontent.com/86923089/209846379-373b97a7-520a-428b-a21b-89ad0844752c.png)